crates/meta/src/storage.rs

```diff
@@ -53,10 +76,74 @@ pub struct FileMetaStorage {
 }
 
 impl FileMetaStorage {
-    pub fn new(root_path: PathBuf) -> Self {
-        Self {
+    pub fn new(root_path: PathBuf) -> Result<Self, BuildError> {
+        if Self::is_empty_directory(root_path.as_path()) {
```

Comment:
Why not checking that the file `.version` exists? This prevents users from mistakes, like adding a file there and then getting a version incompatibility check. Also how about the .DSStore thingy on MacOS?

Edit: Ok i see why, to support version `0`. I wonder if we need to, at the end of the day, the next version of restate is gonna be breaking anyway.

Reply:
I think we need this as long as we want to be able to detect a Restate version which didn't write the `.version` file (Restate <= 0.7). If we say that this is something that we don't want, then we can change it. It is not so much about whether we are introducing breaking changes or not.

Reply:
> If we say that this is something that we don't want

Currently I don't see a point in doing this. Worst case, the former 0.7 user will get an hard error, right?

Reply:
We will do breaking changes anyway between this and the next restate release.

Reply:
> We will do breaking changes anyway between this and the next restate release.

I think this should not be our mindset. Instead I would prefer if we try to maintain compatibility if possible (not at all costs but if possible).

Also if we wouldn't default to format version `0` but instead panic if there is no `.version` file, then it is not really clear for a user how to repair the situation. The user would somehow have to know what was the used format version.

Happy to think about a more robust variant as a follow-up. Right now, I think this mechanism would add value w/o introducing a breaking change for our users.

Reply:
I don't really understand what's the goal here. We know for certain that 0.7 -> 0.8 we'll break storage compatibility, because of some of the planned changes. Are we planning somehow to make this update backward compatible? If yes, will we try to provide tools for this migration? If not, why even try create a robust solution for <= 0.7? Can't we just throw an error in case the `.version` doesn't exist saying "can't find the .version file, seems you're using an old restate data format, please wipe it"? Isn't this actually better than what we have now, that we rely on users reading the changelog?